src/ukify/ukify.py

```diff
@@ -1018,29 +1014,21 @@ def device_make_descriptor(device_type: int, size: int) -> int:
     return (size) | (device_type << 28)
 
 
-def pack_device(offsets: dict[str, int], name: str, compatible: str, chids: list[EFI_GUID]) -> bytes:
+DEVICETREE_DESCRIPTOR = device_make_descriptor(DEVICE_TYPE_DEVICETREE, DEVICE_STRUCT_SIZE)
+
+
+def pack_device(offsets: dict[str, int], name: str, compatible: str, chids: list[uuid.UUID]) -> bytes:
     data = b''
 
-    for data1, data2, data3, data4 in chids:
-        data += struct.pack(
-            '<IIHH8BII',
-            device_make_descriptor(DEVICE_TYPE_DEVICETREE, DEVICE_STRUCT_SIZE),
-            data1,
-            data2,
-            data3,
-            *data4,
-            offsets[name],
-            offsets[compatible],
-        )
+    for chid in chids:
```

Comment:
The uuids should probably also be sorted, because otherwise this could be unreproducible. 
```suggestion
    for chid in sorted(chids):
```

Reply:
This is an array, so shouldn't the order be same same as in the JSON?

Reply:
Looking at the examples I see that name and compatible can actually be the same for different files, so this does depend on the order the files are read in and that is not necessarily sorted.

Reply:
In the end, order doesn't really matter here because of how these UUIDs will be used afterwards in C. It's probably a good idea to remove some duplicates from JSON files though.

Reply:
The order is somewhat important, too, since if the order is non-deterministic, the section in the UKI and therefore the UKI is not reproducible between different invocations. This makes the reproducible builds people sad.